Bring Maui to Microcharts !

Seuleuzeuh opened the pull request:
Implementing Maui plaform for Microcharts.

Adding 2 more supported platforms :  
- [X] Maui
- [X] WinUI

Adding 2 more sample :  
- [X] Maui (including Maui.WinUI)
- [ ] WinUI

Platform tested :  
- [X] Android
- [ ] iOS
- [ ] MacOS
- [ ] Windows

Fix :  
- #189 
- #282 

Troubleshooting :  
I'm unable to launch WinUI project for the moment, need more investigation.

Visuals :
- Android
![Screenshot_1627331106](https://user-images.githubusercontent.com/14994342/127054204-e53a2779-b5dd-4ef7-a67b-4816dfb71189.png)

michael-hawker commented:
Awesome! FYI @brminnick.

By proxy for the Windows part this should enable Microcharts for WinUI 3! FYI @azchohfi

[ShiaSharp already has a WinUI View NuGet](https://www.nuget.org/packages/SkiaSharp.Views.WinUI) so hopefully that should help with the Windows side of this PR. 🎉🎉🎉

eman1986 commented:
Very cool, I'll take a look at this tonight

eman1986 commented:
@Seuleuzeuh This looks good, should I assume you're still working on this since it's WIP?

Seuleuzeuh commented:
> @Seuleuzeuh This looks good, should I assume you're still working on this since it's WIP?

Yes I will work on it again today

Seuleuzeuh commented:
> Awesome! FYI @brminnick.
> 
> By proxy for the Windows part this should enable Microcharts for WinUI 3! FYI @azchohfi
> 
> [ShiaSharp already has a WinUI View NuGet](https://www.nuget.org/packages/SkiaSharp.Views.WinUI) so hopefully that should help with the Windows side of this PR. 🎉🎉🎉

This PR will include WinUI support too @michael-hawker :smile:

Seuleuzeuh commented:
@eman1986 i'm unable to test it on WinUI (bad VS 2022 installation) and iOS/MacOs version (error with Pair to Mac).
I don't know if you want to merge it anyway ?

I've updated all the Skia dependencies to the 2.88.0-preview.61, necessary for Maui. We'll need to make update each preview version of .Net-6 and finally when the RC will come.
I can handle that update part every new version.

I've attached all the nuspec file in the Solution (Solution items) for quicker update.

I've updated the readme and adding specific instruction for maui install.

Without the mac connection i can't generate a proper usable nuget, i'm working on it, but for now i didn't have luck.
If you are able to generate the Maui-preview nuget it would be great !

Keep me inform :wink:

eman1986 commented:
I can give it a try, with MAUI being still preview as is our 1.0 codebase I'm ok with merging it in, .NET 6 is right around the corner so we should be fine.

eman1986 commented:
We're not really looking to support web things at this time

LarsBehl commented:
I'm currently working on a project based on MAUI and would like to use the lib to display some graphs. Any updates on the state of this PR?

eman1986 commented:
There's currently a merge conflict that needs to be fixed.

eman1986 commented:
> I'm currently working on a project based on MAUI and would like to use the lib to display some graphs. Any updates on the state of this PR?

Once the PR is fixed up, I can get it merged in. It'd be in a preview release so bare that in mind.

LarsBehl commented:
Thanks. I'll keep in mind that it still is in preview.

pavelbelousov commented:
Hello, any updates on the merge status?

I'm starting a new MAUI project and would appreciate having the possibility to use this awesome library.

jBijsterboschNL commented:
Same here, would like to start a next project with .NET MAUI, but I'm only able to work with .NET MAUI if this great control set is ported to .NET 6! Hopefully a preview release can be distributed soon! :)

Ghostbird commented:
Is this dead? I'm porting my app to MAUI and .NET 7, but it seems I'll have to find another library.

acaliaro commented:
> Is this dead? I'm porting my app to MAUI and .NET 7, but it seems I'll have to find another library.

there is [AlohaKit](https://github.com/jsuarezruiz/AlohaKit.Controls) that has some controls

eman1986 commented:
Sadly this lost progress, but if someone could help out with it, it'd be a major help. I unfortunately have shifted to doing native mobile development and no longer have xamarin on my machine.

Ghostbird commented:
@eman1986 Thanks for the heads up. @acaliaro Thanks for the tip. I'm going to have a look at it. I don't have time to contribute outside of work. I was trying to port it to MAUI with feature parity, however as the charts feature isn't even officially supported in our app, I think we'll drop it, if we can't find a quick replacement library. If we get a paying customer for it, I might actually get an opportunity to contribute.

Eilon commented:
In case anyone is interested, I'm trying to see if I can port this to .NET MAUI 6.0 with some updated code.

Eilon commented:
WinUI and Android working in my branch:

Here's WinUI:

<img width="584" alt="image" src="https://user-images.githubusercontent.com/202643/214650090-33f70331-d9e7-4911-82ee-be816b45e3d4.png">

And Android:

<img width="267" alt="image" src="https://user-images.githubusercontent.com/202643/214650398-819f816b-95c2-4eac-bdef-2515c71bbad8.png">

I'll pull out my Mac a bit later to test iOS/MacCatalyst.

Eilon commented:
Unfortunately mixed status for iOS/MacCatalyst. On both platforms the MAUI sample app loads, but it then immediately hangs and I can't click on anything in the app, so I have to kill it.

MacCatalyst (hung):

<img width="681" alt="image" src="https://user-images.githubusercontent.com/202643/214655313-80c6e724-1dc9-4dda-8f0e-00a98ca3562f.png">

iOS (hung in Simulator):

<img width="240" alt="image" src="https://user-images.githubusercontent.com/202643/214655373-b9df0bcb-06fb-4037-85f2-18998a753131.png">

Eilon commented:
My PR that updates this PR is available here: https://github.com/microcharts-dotnet/Microcharts/pull/325
